pymonetdb/mapi.py

```diff
@@ -166,7 +165,7 @@ def connect(self, database: str, username: str, password: str, language: str,  #
         self.database = database
         self.language = language
         self.unix_socket = unix_socket
-        self.handshake_options = handshake_options or []
+        self.handshake_options = handshake_options or (lambda b, opts: [])
```

Comment:
so handshake_options is an iterable of callables? maybe add that to the type signature, since it is unusual.

Reply:
It was a function that took a number (used to be a boolean) and a dictionary and returned a list of handshake options. I've attempted to clean up the code.